interview-questions.md

```diff
@@ -8,30 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
+
 3. What is the difference between state values and props in React?
 
 Your answer:
 
+State values an element of a type of variable in React called useState that is perfect for storing information that will be updated on the webpage. The useState is declared equivalent to a state value and a state function, and given a starting point for the state value. The state value is then updated by invoking the state function. 
+
+Props is passed as a parameter in a nested function of a React app. The props parameter operates as a pipeline passing information upstream from the nested component to the DOM. The state value can be passed in the nested component function as the argument in place of the parameter props. 
+
 Researched answer:
 
+State is a data structure used to store data in a React component, that can then be accessed anywhere within the React component. State is created by the useState method, which defines a variable, the varibales defualt value, and a function that will be used to update the variable. useState is an example of a React hooks. Hooks are a collection of built-in methods in React that allow developers to access reusable features. 
+
+Props is a specialized tool in React that allows the transfer of both both data and behavior between components. Props is unidirectional, passing information downstream to nested components when the component is invoked. Because props's data structure is an object, it can be used to pass as much information via key value pairs as needed. Data communicated through props often comes from the state value. While state is made to change and update, props is "read-only" and do not change, but pass the updated state through the pipeline from one component down to the next. 
+
+
 4. STRETCH: Which is the difference between a div tag and a span tag?
 
 Researched answer:
 
+A div tag is know as a block-level element, meaning it spans horizontally across the entirety of the webpage and browsers automatically put it on a new line of the webpage and add some margin on the top and bottom. This makes it useful to wrap whole sections of a webpage, and group together large amounts of content. Other examples of block-level elements are the <p> tag and <footer> tag. The span tag is an inline element, meaning the browser does not put the contnt on a new line of the webpage, and only takes up the necessary width on the webpage. The span tag is useful for grouping small sections of text or images. Other examples of inline elemenmts are the <button> tag and <img> tag.  
+
```

Comment:
Fantastic work on these answers!  I really like that you fully expanded after researching. Beautiful! You're killing it!